astroquery/query.py

```diff
@@ -102,20 +107,30 @@ def hash(self):
         return self._hash
 
     def request_file(self, cache_location):
-        fn = os.path.join(cache_location, self.hash() + ".pickle")
+        fn = cache_location.joinpath(self.hash() + ".pickle")
         return fn
 
-    def from_cache(self, cache_location):
+    def from_cache(self, cache_location, cache_timeout):
         request_file = self.request_file(cache_location)
         try:
-            with open(request_file, "rb") as f:
-                response = pickle.load(f)
-            if not isinstance(response, requests.Response):
+            if cache_timeout is None:
+                expired = False
+            else:
+                current_time = datetime.utcnow()
+                cache_time = datetime.utcfromtimestamp(request_file.stat().st_mtime)
+                expired = current_time-cache_time > timedelta(seconds=cache_timeout)
+            if not expired:
+                with open(request_file, "rb") as f:
+                    response = pickle.load(f)
```

Comment:
I wonder how much problem it will be to picking into the same location when one uses multiple python versions? Could we make this robust for python/numpy version or do we trust that the expiration will likely cover the problematic cases?

Reply:
I think it should be ok. Pickle was always incompatible between python 2 and 3, but we only support python 3 now. I've updated it to explicitly specify the protocol as 4, which is available under all python versions we support. I think that's probably the best we can do without using something other than pickle.